Enable FC tests

tugsbayasgalan opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #72479
* #72474

Differential Revision: [D34057581](https://our.internmc.facebook.com/intern/diff/D34057581)

tugsbayasgalan committed:
Enable FC tests

[ghstack-poisoned]

tugsbayasgalan commented:
@tugsbayasgalan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34057581).

tugsbayasgalan committed:
Update on "Enable FC tests"


Differential Revision: [D34057581](https://our.internmc.facebook.com/intern/diff/D34057581)

[ghstack-poisoned]

tugsbayasgalan commented:
@tugsbayasgalan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34057581).

tugsbayasgalan committed:
Update on "Enable FC tests"


Differential Revision: [D34057581](https://our.internmc.facebook.com/intern/diff/D34057581)

[ghstack-poisoned]

tugsbayasgalan commented:
@tugsbayasgalan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34057581).

tugsbayasgalan committed:
Update on "Enable FC tests"


Differential Revision: [D34057581](https://our.internmc.facebook.com/intern/diff/D34057581)

[ghstack-poisoned]

tugsbayasgalan commented:
@tugsbayasgalan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34057581).

mruberry commented on test/forward_backward_compatibility/check_forward_backward_compatibility.py:
```diff
@@ -31,13 +31,18 @@
 ]
 
 # WARNING: Operators included in this list indefinitely bypass all BC and FC schema checks.
 # This is almost certainly NOT what you want to do. See note above.
```
Which note does this refer to? The note should be given a name

mruberry commented on test/forward_backward_compatibility/check_forward_backward_compatibility.py:
```diff
@@ -31,13 +31,18 @@
 # ]
 #
 # NB: function name DOES NOT include overload name!
```
We should also point to our BC/FC policy

mruberry commented on test/forward_backward_compatibility/check_forward_backward_compatibility.py:
```diff
@@ -76,9 +81,50 @@
-def compile_temp_allow_list():
+# Same thing as INDEFINITE_BC_ALLOW_LIST
+# but for FC changes.
+INDEFINITE_FC_ALLOW_LIST = [
```
What's dangerous about adding an operator to this list?

mruberry commented:
This stack looks OK but the lint and test failure appear real -- what's up with that?

tugsbayasgalan committed:
Update on "Enable FC tests"


Differential Revision: [D34057581](https://our.internmc.facebook.com/intern/diff/D34057581)

[ghstack-poisoned]

tugsbayasgalan commented:
@tugsbayasgalan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34057581).

tugsbayasgalan committed:
Update on "Enable FC tests"


Differential Revision: [D34057581](https://our.internmc.facebook.com/intern/diff/D34057581)

[ghstack-poisoned]

tugsbayasgalan committed:
Update on "Enable FC tests"


Differential Revision: [D34057581](https://our.internmc.facebook.com/intern/diff/D34057581)

[ghstack-poisoned]
